OSDOCS-11988: 4.16.12 RN (#81983)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.16.12
[id="ocp-4-16-12_{context}"]

> === RHSA-2024:6632 - {product-title} {product-version}.12 bug fix update

Reviewer: This is a bug fix and security update.

> === RHSA-2024:6632 - {product-title} {product-version}.12 bug fix update

Reviewer: This is a bug fix and security update.

Issued: 17 September 2024

> {product-title} release {product-version}.11 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:6632[RHSA-2024:6632] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:6635[RHBA-2024:6635] advisory.

Reviewer 2: Just want to make sure..the title says {product-version}.12 update, and this line 3360 says {product-version}.11. Should these match or not?

Author: Yes! good catch!

> {product-title} release {product-version}.12 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:6632[RHSA-2024:6632] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:6635[RHBA-2024:6635] advisory.

Reviewer: Doesn't look like this shipped live yet.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.16.12 --pullspecs
----

[id="ocp-4-16-12-enhancements_{context}"]
==== Enhancements

The following enhancements are included in this z-stream release:

[id="ocp-4-16-12-https-support-TransferProtocolTypes"]
===== Supporting HTTPS for TransferProtocolTypes in Redfish APIs

> * TLS can be enabled for communication between ironic and the baseboard management controller (BMC) in the bootstrap phase of the install process by adding 'disableVirtualMediaTLS: false' to the Provisioning CR file created on disk by the installer. (link:https://issues.redhat.com/browse/OCPBUGS-39468[*OCPBUGS-39468*])

Reviewer 2: ```suggestion
* TLS can be enabled for communication between ironic and the baseboard management controller (BMC) in the bootstrap phase of the install processes by adding 'disableVirtualMediaTLS: false' to the Provisioning CR file created on disk by the installer. (link:https://issues.redhat.com/browse/OCPBUGS-39468[*OCPBUGS-39468*])
```

[id="ocp-4-16-12-kubernetes-1-29-8"]
===== Updating to Kubernetes version 1.29.8

* This release contains the changes that come from the update to Kubernetes version 1.29.8. (link:https://issues.redhat.com/browse/OCPBUGS-39015[*OCPBUGS-39015*])

[id="ocp-4-16-12-redirecting-git-advanced"]
===== Redirecting on web console with Edit Source Code

> * There are 2 option in the _Git Advanced_ section of the web console: one option is to add a branch, tag, or commit ID and other option is to add the context directory. With this release, if you add the context directory of a particular branch, tag, or commit ID, you are redirected to that directory by selecting the *Edit source code* icon. If a branch, tag, or commit ID is not entered, you are redirected to the base url as previously expected. (link:https://issues.redhat.com/browse/OCPBUGS-38914[*OCPBUGS-38914*])

Reviewer 2: ```suggestion
* There are two options in the _Git Advanced_ section of the web console: one option is to add a branch, tag, or commit ID and other option is to add the context directory. With this release, if you add the context directory of a particular branch, tag, or commit ID, you are redirected to that directory by selecting the *Edit source code* icon. If a branch, tag, or commit ID is not entered, you are redirected to the base url as previously expected. (link:https://issues.redhat.com/browse/OCPBUGS-38914[*OCPBUGS-38914*])
```

Reviewer 2: since you use "one option" and "the other", its more consistent to use "two" instead of the numeral here: https://www.ibm.com/docs/en/ibm-style?topic=measurement-numerals-versus-words

Reviewer 2: Also option->options

[id="ocp-4-16-12-bug-fixes_{context}"]
==== Bug fixes

* Previously, when a large number of secrets in a cluster were fetched in a single call, the API timed out and the CCO threw an error and then restarted. With this release, the CCO pulls the secret list in smaller batches of 100 and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41234[*OCPBUGS-41234*])

* Previously, Operator Lifecycle Manager (OLM) catalog source pods did not recover from node failure if the `registryPoll` field was `none`. With this release, OLM CatalogSource registry pods recover from cluster node failures and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41217[*OCPBUGS-41217*])

> * Previously, the Cluster ingress operator logged non-existent updates. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-39324[*OCPBUGS-39324*])

Reviewer 2: Does Cluster Ingress Operator need to be capitalized?

* Previously, the installation program failed to install an {product-title} cluster in the `eu-es` (Madrid, Spain) region on a {ibm-power-server-title} platform that was configured as an `e980` system type. With this release, the installation program no longer fails to install a cluster in this environment. (link:https://issues.redhat.com/browse/OCPBUGS-38502[*OCPBUGS-38502*])

> * Previously, the ingress controller `Degraded` status would not set because of an issue with the `CanaryRepetitiveFailures` condition transition time. With this release, the condition transition time is only updated when the condition status changes, instead of when the message or reason are the only changes. (link:https://issues.redhat.com/browse/OCPBUGS-39323[*OCPBUGS-39323*])

Reviewer 2: ```suggestion
* Previously, the Ingress Controller `Degraded` status would not set because of an issue with the `CanaryRepetitiveFailures` condition transition time. With this release, the condition transition time is only updated when the condition status changes, instead of when the message or reason are the only changes. (link:https://issues.redhat.com/browse/OCPBUGS-39323[*OCPBUGS-39323*])
```

Reviewer 2: https://redhat-documentation.github.io/supplementary-style-guide/#ingress-controller

* Previously, an `AdditionalTrustedCA` field that was specified in the Hosted Cluster image configuration was not reconciled into the openshift-config namespace as expected and the component was not available. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-39293[*OCPBUGS-39293*])

* Previously, an installer regression issue caused problems with Nutanix cluster deployments using the Dynamic Host Configuration Protocol (DHCP) network. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-38956[*OCPBUGS-38956*])

> * Previously, a rare condition caused the CAPV session to time out unexpectedly. With this release, the *Keep Alive* support is disabled in later versions of CAPV and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-38822[*OCPBUGS-38822*])

Reviewer 2: ```suggestion
* Previously, a rare condition caused the CAPV session to time out unexpectedly. With this release, the *Keep Alive* support is disabled in later versions of CAPV, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-38822[*OCPBUGS-38822*])
```

Reviewer 2: need a comma here after this conjunction and before this independent clause

> * Previously, the version number text in the updates graph on the Cluster Settings appeared as black text on a dark background while viewing the page using Firefox in dark mode. With this update, the text appears as white text. (link:https://issues.redhat.com/browse/OCPBUGS-38822[*OCPBUGS-38424*])

Reviewer 2: ```suggestion
* Previously, the version number text in the updates graph on the *Cluster Settings* appeared as black text on a dark background while viewing the page using Firefox in dark mode. With this update, the text appears as white text. (link:https://issues.redhat.com/browse/OCPBUGS-38822[*OCPBUGS-38424*])
```

Reviewer 2: This seems like a UI item so it should be bolded if it is

> * Previously, proxying for Operators that run in the control plane of a Hypershift cluster was performed through proxy settings on the konnectivity agent pod that runs in the data plane. As a result, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with {rh-short}, IDP communication through https/http should be proxied, but LDAP communication should not be proxied. With this release, how proxy is handled in hosted clusters is changed to invoke the proxy in the control plane via konnectivity-https-proxy and konnectivity-socks5-proxy, and to stop proxying traffic from the konnectivity agent. (link:https://issues.redhat.com/browse/OCPBUGS-38062[*OCPBUGS-38062*])

Reviewer 2: ```suggestion
* Previously, proxying for Operators that run in the control plane of a HyperShift cluster was performed through proxy settings on the konnectivity agent pod that runs in the data plane. As a result, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with {rh-short}, IDP communication through https/http should be proxied, but LDAP communication should not be proxied. With this release, how proxy is handled in hosted clusters is changed to invoke the proxy in the control plane via `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and to stop proxying traffic from the konnectivity agent. (link:https://issues.redhat.com/browse/OCPBUGS-38062[*OCPBUGS-38062*])
```

Reviewer 2: - Searching the repo, I think HyperShift the S is capitalized.
- do these `konnectivities` need backticks? It seems like it to me
